consumer/consumer.go

```diff
@@ -92,24 +95,33 @@ func New(brokerCfg broker.Configuration, storage storage.Storage) (*KafkaConsume
 	return NewWithSaramaConfig(brokerCfg, storage, DefaultSaramaConfig, brokerCfg.SaveOffset)
 }
 
-// NewWithSaramaConfig constructs new implementation of Consumer interface with custom sarama config
-func NewWithSaramaConfig(
-	brokerCfg broker.Configuration,
-	storage storage.Storage,
-	saramaConfig *sarama.Config,
-	saveOffset bool,
-) (*KafkaConsumer, error) {
+func createKafkaConsumer(brokerCfg broker.Configuration, saramaConfig *sarama.Config) (sarama.Client, sarama.Consumer, []int32, error) {
 	client, err := sarama.NewClient([]string{brokerCfg.Address}, saramaConfig)
 	if err != nil {
-		return nil, err
+		return nil, nil, nil, err
```

Comment:
this reminds me code (constructors) in ccx-data-pipeline :)

Reply:
Haha, yeah, I thought of that too when I wrote it. I believe these return of many `nil`s is no longer in the refactored code now, thankfully.